src/js/visual/image.js

```diff
@@ -420,6 +420,13 @@ ripe.Image.prototype.update = async function(state, options = {}) {
             curve: curve
         });
 
+        // if there are message events in initials builder ctx, dispatch them
+        if (initialsSpec.ctx && initialsSpec.ctx.messages && initialsSpec.ctx.messages.length) {

```

Comment:
This should be done when running the initials builder to be aligned with the server side.
I'm not sure this is the "perfect" place to position this code

Reply:
Initials builder runs here https://github.com/ripe-tech/ripe-sdk/blob/e3bc605b7c243305bd65a7ef1019b303382be09a/src/js/visual/image.js#L349

`initialsSpec` holds the result of running the initials builder. I will move it closer to the call...